packages/teleport/src/AppLauncher/useAppLauncher.ts

```diff
@@ -62,7 +63,9 @@ function resolveRedirectUrl(params: UrlLauncherParams) {
       if (params.arn) {
```

Comment:
Should this be `if (arn)` instead?

Reply:
I think so. Although I'm not quite sure the context just yet. Similar question to this line
```js
  // state value received: create new session for the target app
  if (arn) {
    params.arn = arn;
  }
```
I'm not 100% sure why we would pull arn from params, and then if it exists, put it back in. Any context @rudream ?

Reply:
This looks like there are two different ways of getting the value for `arn`. In the `UrlLauncherParams` it's pulling a value called `arn` but on line 50 above it's getting one called `awsrole`. So My guess would be that this is to support a fallback if the value in `params` is not defined. And this would be why it gets set again because it's preferring the value from `arn` not the, possibly legacy, `awsrole`. This is is all speculation ;) 

```
export interface UrlLauncherParams {
  fqdn: string;
  clusterId?: string;
  publicAddr?: string;
  arn?: string;
}
```